groth16/src/generator/generic.rs

```diff
@@ -35,6 +36,7 @@ where
 
 /// This is our assembly structure that we'll use to synthesize the
 /// circuit into a QAP.
+#[derive(Clone, Debug, CanonicalSerialize, CanonicalDeserialize)]
```

Comment:
This is a data structure used to synthesize constraints, but doesn't contain the resulting public keys. I don't expect that this should ever be serialized to disk, right?